src/management-system-v2/components/modeler-share-modal-option-embed-in-web.tsx

```diff
@@ -0,0 +1,84 @@
+import useModelerStateStore from '@/lib/use-modeler-state-store';
+import { useEffect, useRef, useState } from 'react';
+
+import { CopyOutlined } from '@ant-design/icons';
+import { Button, message } from 'antd';
+import TextArea from 'antd/es/input/TextArea';
+
+const ModelerShareModalOptionEmdedInWeb = () => {
+  const [bmpnXML, setBpmnXml] = useState('');
+  const modeler = useModelerStateStore((state) => state.modeler);
+  const codeSection = useRef(null);
+
+  const handleCopyCodeSection = async () => {
+    const codeToEmbed = codeSection.current?.resizableTextArea.textArea.value;
+    if (codeToEmbed) {
+      await navigator.clipboard.writeText(codeToEmbed);
+      message.success('Code copied successfully');
+    }
+  };
+
+  const getXML = async () => {
+    if (modeler) {
+      const { xml } = await modeler.saveXML({ format: true });
+      if (xml) setBpmnXml(xml);
+    }
+  };
+
+  useEffect(() => {
+    getXML();
+  }, []);
+
+  return (
+    <>
+      <div style={{ marginBottom: '100px' }}>
+        <Button
+          icon={<CopyOutlined />}
+          style={{ border: '1px solid black', float: 'right' }}
+          onClick={handleCopyCodeSection}
+        >
+          Copy Code
+        </Button>
+      </div>
+      <div className="code">
+        <TextArea
+          rows={10}
+          style={{ backgroundColor: 'rgb(245,245,245)' }}
+          value={`<div class="canvas">
+            <div id="js-canvas"></div>
```

Comment:
I think due to the fact that #183 talks about an iframe for the embed  button the functionality should create some kind of partial embedding of our MS focused on the modeler with the specific process being open and not editable.
So if a user chooses to embed a process then they will always see the current state of the process instead of the state at the time when the button was clicked.